### ISSUE TITLE
Redesign date selection in search & filters modal to be type-to-edit

### ISSUE THREAD

**Issue by ryanfchase** [hackforla/311-data#1818]
### Overview
We need to improve the usability of the date selection component of the search and filters modal, since users are reporting it does not behave the way they expect it to.

### Action Items

- [ ] Address the problem overview with possible design solution(s) in a comment
- [ ] Complete Design Iterations section below
- [ ] Document user interaction in Figma

### Resources

V1.2 Moderated Usability Test
- [October '23 slide on wanting to type-to-edit date range](https://docs.google.com/presentation/d/1NBMnS8_znCnYNDNNQrN2Jh5GQZhZmX56rtHME-EewBY/edit#slide=id.g29f7a872fc9_0_100)
- [October '23 slide on users not noticing calendar dropdown icon](https://docs.google.com/presentation/d/1NBMnS8_znCnYNDNNQrN2Jh5GQZhZmX56rtHME-EewBY/edit#slide=id.g29f7a872fc9_0_123)
- [August '24 slide on wanting to type-to-edit date range](https://docs.google.com/presentation/d/1w1YpzC-sM7X1OOQ70f5Ey8h6Q0yOoTlC/edit#slide=id.p13)

Comments on V1.2 August Presentation by UIUX Design Team Members
- [link to google doc with comments](https://docs.google.com/document/d/1-5-nPHXsoaMXsKtObEnoIp4eQFCF9DN0i6Lk2IhHBxE/edit)

---

### Design Iterations

**Please move ticket between `In Progress` and `In Review` to assist PM team**

<details><summary>Iteration 1</summary>
<p>

Link to notes: `REPLACE WITH COMMENT URL`

`REPLACE WITH SCREENSHOT UPLOAD`

</p>
</details> 

---

### Hand Off Materials

Figma Section Name: `REPLACE WITH SECTION NAME`

<details><summary>Before Screenshot</summary>
<p>

`REPLACE WITH SCREENSHOT UPLOAD`

</p>
</details> 

<details><summary>After Screenshot (Finalized)</summary>
<p>

`REPLACE WITH SCREENSHOT UPLOAD`

</p>
</details> 

### Designer Resources

<details><summary>Iteration Dropdown Copy/Paste</summary>
<p>

```
<details><summary>Iteration X</summary>
<p>

Link to notes: `REPLACE WITH COMMENT URL`

`REPLACE WITH SCREENSHOT UPLOAD`

</p>
</details> 
```

</p>
</details> 

<details><summary>Instructions for Engineering Hand Off</summary>
<p>

To Start Engineering Hand Off...
1. Ensure all Hand Off Materials are filled in
3. Add the "ready for dev lead" label
4. Leave a comment saying "This ticket is ready for engineering hand off." 

</p>
</details>

**Comment by ryanfchase** [comment]
Ready for design lead input. Please add:
- any further required information
- any possible blockers
- remove `Size: Missing` and add appropriate size

**Comment by Joy-Truex** [comment]
@allisonjeon Can you please review the information on this ticket and address the points ryan has commented and add the size and determine if it should be combined with [#1600 ](https://github.com/hackforla/311-data/issues/1600)

**Comment by ryanfchase** [comment]
@Joy-Truex  @allisonjeon PMs are trying to estimate Post-Launch tickets. Can you provide a size estimation? Thanks

**Comment by ryanfchase** [comment]
Size has been updated. Relaying this change to our Aug Action Steps Spreadsheet:

- Size: 2pt

**Comment by ryanfchase** [comment]
🟢 Ticket has been prioritized and is ready to be picked up

**Comment by danachou** [comment]
Ideated on the solution, found references, started initial wireframes.

**Comment by Joy-Truex** [comment]
potential reference: [Date-Input Form Fields: UX Design Guidelines
](https://www.nngroup.com/articles/date-input/)

**Comment by danachou** [comment]
@ryanfchase  do you know of some common date ranges that users typically want to enter.

**Comment by ryanfchase** [comment]
@danachou The app currently pre-loads the latest 2 weeks of data (it _should_, it's broken right now ☹). Otherwise, I don't think we have research that would give us data on that. @Joy-Truex can we think of a reason that we might need to obtain this?

**Comment by danachou** [comment]
> [@danachou](https://github.com/danachou) The app currently pre-loads the latest 2 weeks of data (it _should_, it's broken right now ☹). Otherwise, I don't think we have research that would give us data on that. [@Joy-Truex](https://github.com/Joy-Truex) can we think of a reason that we might need to obtain this?

@ryanfchase I was trying to understand whether we need to design for users selecting large date ranges (e.g., months or years) or if we can optimize primarily for smaller ranges like the past 2 weeks. I will look around in user interview data for that!

**Comment by cottonchristopher** [comment]
Due to the fact that we are now implementing a "blank" map, should we re-think the action items/necessity of prefilled data ranges? Asking because perhaps we can remove this action item. @Joy-Truex 
> - [ ] PM Requirement: remove "Last 3 Months" from the date range dropdown and replace with "Custom Date Range"

**Comment by Joy-Truex** [comment]
@cottonchristopher Yes the user should be entering in the date range and there should not be specified or prefilled ranges.

**Comment by Joy-Truex** [comment]
@danachou was requesting this information--

> @ryanfchase I was trying to understand whether we need to design for users selecting large date ranges (e.g., months or years) or if we can optimize primarily for smaller ranges like the past 2 weeks. I will look around in user interview data for that!

for the purpose of determining the size of the calendar pop out, if it should be more compact or larger depending on the custom date ranges users are trying to enter.

**Comment by Joy-Truex** [comment]
@danachou please leave a detailed update of where you are at in this design ticket process as we prepare to pause work for the month of april. thank you!! 😄

**Comment by danachou** [comment]
Updated the date selection from selection to type-to-edit. 
Did a few iterations of multiple calendar layouts and with team feedback decided on the single calendar format. 
Added text entry fields to accommodate two dates (start and end dates). 

![Image](https://github.com/user-attachments/assets/028815ce-cce2-4cc6-94ad-2576d6c4c95b)

![Image](https://github.com/user-attachments/assets/0bb84b20-3abb-41c7-a458-a16e91a6c68d)

![Image](https://github.com/user-attachments/assets/47215878-0845-48bb-a976-c4e54abbabb5)

**Comment by Joy-Truex** [comment]
Will be ready for final review and handoff prep once we return from TDM hiatus.

**Comment by Joy-Truex** [comment]
After discussing with the design team in our breakout, we have determined that the only relevant action item is to remove the date chips (see attachment) which has already been addressed by a previous ticket.

<img width="151" height="424" alt="Image" src="https://github.com/user-attachments/assets/b7a20824-fb8d-453f-bf3a-cfd318dd63e3" />
